internal/objects/validator_test.go

```diff
@@ -210,7 +210,63 @@ func TestUserChanges(t *testing.T) {
 				},
 			},
 		},
-	}, {
+	},{
+		name: "Deny metadata changes to HNC annotations in propagated objects",
+		fail : true,
+		oldInst: &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "v1",
+				"kind": "Pod",
+				"metadata": map[string]interface{}{
+					"labels": map[string]interface{}{
+						api.LabelInheritedFrom: "foo",
+					},
+				},
+			},
+		},
+		inst: &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "v1",
+				"kind": "Pod",
+				"metadata": map[string]interface{}{
+					"labels": map[string]interface{}{
+						api.LabelInheritedFrom: "foo",
+					},
+					"annotations": map[string]interface{}{
+						api.AnnotationPropagatePrefix + "/select": "abc",
+					},
+				},
+			},
+		},
+	}, 
+	{
+		name: "Deny metadata changes to HNC labels in propagated objects",
+		fail : true,
+		oldInst: &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "v1",
+				"kind": "Pod",
+				"metadata": map[string]interface{}{
+					"labels": map[string]interface{}{
+						api.LabelInheritedFrom: "foo",
+					},
+				},
+			},
+		},
+		inst: &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "v1",
+				"kind": "Pod",
+				"metadata": map[string]interface{}{
+					"labels": map[string]interface{}{
+						api.LabelInheritedFrom: "foo",
+						api.AnnotationPropagatePrefix + "/select": "abc",
```

Comment:
It's a bit weird to use a variable called "Annotation" in a set of labels... - how about just `api.MetaGroup + "/foo"?